diff --git a/kernel/irq/ipi.c b/kernel/irq/ipi.c
index 1a9abc1..81d8c5f 100644
--- a/kernel/irq/ipi.c
+++ b/kernel/irq/ipi.c
@@ -124,8 +124,9 @@ int irq_destroy_ipi(unsigned int irq, const struct cpumask *dest)
 		return -EINVAL;
 
 	domain = data->domain;
-	if (WARN_ON(domain == NULL))
+	if (WARN_ON(domain == NULL)) {
 		return -EINVAL;
+    }
 
 	if (!irq_domain_is_ipi(domain)) {
 		pr_warn("Trying to destroy a non IPI domain!\n");
