--- clang.orig/CMakeLists.txt
+++ clang/CMakeLists.txt
@@ -101,7 +101,7 @@
   include(AddLLVM)
   include(TableGen)
   include(HandleLLVMOptions)
-  include(VersionFromVCS)
+  include(VersionFromVCS OPTIONAL RESULT_VARIABLE HAVE_VERSION_FROM_VCS)
 
   set(PACKAGE_VERSION "${LLVM_PACKAGE_VERSION}")
 
@@ -217,7 +217,7 @@
 option(CLANG_APPEND_VC_REV
   "Append the version control system revision id to clang version spew" OFF)
 
-if(NOT SVN_REVISION)
+if(NOT SVN_REVISION AND HAVE_VERSION_FROM_VCS)
   # This macro will set SVN_REVISION in the parent scope
   add_version_info_from_vcs(VERSION_VAR)
 endif()
